### PR TITLE
Add new unittest file for basic:label module

### PR DESCRIPTION
**Pull request by devdekunle**
The new test files contains testcases for the basic:label module. it tests the three functions label_ops_set, label_ops_pre and label_ops_post

### CODE REVIEW

File: src/test/test-label.c [hunk 1]
```diff
@@ -0,0 +1,74 @@
+#include <errno.h>
```
Comment (bluca) [inline src/test/test-label.c:3]: SPDX license line is missing

File: src/test/test-label.c [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+#include <stddef.h>
+#include <string.h>
+
+#include "label.h"
+#include "tests.h"
+#include "string-util.h"
```
Comment (bluca) [inline src/test/test-label.c:10]: sort alphabetically

File: src/test/test-label.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+TEST(label_ops_post) {
+
+    assert_se(label_ops_post(1, "abcd") == 0);
+    test_label_ops.post = post_labelling_func;
+    assert_se(label_ops_post(1, "abc") == 0);
+    assert_se(label_ops_post(1, "/restricted_directory") == -1);
+    label_ops_set(NULL);
+}
+
+DEFINE_TEST_MAIN(LOG_INFO)
```
Comment (bluca) [inline src/test/test-label.c]: new line at the end of file missing

Comment (devdekunle) [reply src/test/test-label.c]: Thank you @bluca for the corrections. I have implemented the changes

**Comment by devdekunle** [comment]
I've implemented the changes @bluca Kindly review

**Comment by bluca** [comment]
The test is failing, also rebase squash fixups commits

**Comment by bluca** [comment]
Still failing:

```
stderr:
/* test_label_ops_post */
/* test_label_ops_pre */
Assertion 'label_ops_pre(1, "/restricted_directory", 0755) == -1' failed at src/test/test-label.c:47, function test_label_ops_pre(). Aborting.
```

File: src/test/test-label.c [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+}
+
+TEST(label_ops_post) {
+
+    assert_se(label_ops_post(1, "abcd") == 0);
+    test_label_ops.post = post_labelling_func;
+    assert_se(label_ops_post(1, "abc") == 0);
+    label_ops_set(NULL);
+}
```
Comment (poettering) [inline src/test/test-label.c:90]: please follow coding style, i.e. 8space indentation.

https://systemd.io/CODING_STYLE/

File: src/test/test-label.c [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+#include <errno.h>
+#include <stddef.h>
+#include <string.h>
+
+#include "label.h"
+#include "string-util.h"
+#include "tests.h"
+
+
```
Comment (poettering) [inline src/test/test-label.c:14]: spurious double empty line

File: src/test/test-label.c [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+int pre_labelling_func(int dir_fd, const char *path, mode_t mode);
+int post_labelling_func(int dir_fd, const char *path);
+LabelOps test_label_ops = {
+        .pre = NULL,
+        .post = NULL
+    };
```
Comment (poettering) [inline src/test/test-label.c]: static const

File: src/test/test-label.c [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+#include <stddef.h>
+#include <string.h>
+
+#include "label.h"
+#include "string-util.h"
+#include "tests.h"
+
+
+int pre_labelling_func(int dir_fd, const char *path, mode_t mode);
+int post_labelling_func(int dir_fd, const char *path);
```
Comment (poettering) [inline src/test/test-label.c]: why these prototypes?

File: src/test/test-label.c [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    /*attempt to reset label_ops when already set*/
+    assert_se(label_ops_set(&test_label_ops) == -EBUSY);
+
+}
+
+int pre_labelling_func(int dir_fd, const char *path, mode_t mode) {
+    /*sample pre_labelling task*/
+    if (dir_fd < 0 || !path || mode == 0)
+        return -1;
```
Comment (poettering) [inline src/test/test-label.c]: do not make up errors like -1. Return negative errno style errors, i.e. -EINVAL or so. See coding style

File: src/test/test-label.c [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+
+    /*assume a case where a specific label isn't allowed*/
+    if (streq(path, "/restricted_directory"))
```
Comment (poettering) [inline src/test/test-label.c]: path_equal()

File: src/test/test-label.c [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+        return -1;
```
Comment (poettering) [inline src/test/test-label.c]: as above

File: src/test/test-label.c [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    assert_se(label_ops_pre(1, "/restricted_directory", 0755) == -1);
+    assert_se(label_ops_pre(1, "abcd", 0) == -1);
+
+}
+
+int post_labelling_func(int dir_fd, const char *path) {
+    /*sample post_labelling task*/
+
+    if (dir_fd < 0 || !path)
+    return -1;
```
Comment (poettering) [inline src/test/test-label.c]: indentation is not right

**Review by poettering — CHANGES_REQUESTED** [review]

File: src/test/test-label.c (hunk 8 above)
Comment (devdekunle) [reply src/test/test-label.c]: Thank you @poettering I have refactored the code by removing the prototypes and adjusting indentations

**Pull request closed without merging**